ocaml/xapi/xapi_ha.ml

```diff
@@ -78,36 +78,51 @@ let local_failover_decisions_are_ok () =
   try not (bool_of_string (Localdb.get Constants.ha_disable_failover_decisions))
   with _ -> true
 
-(** Since the liveset info doesn't include the host IP address, we persist these ourselves *)
-let write_uuid_to_ip_mapping ~__context =
-  let table =
-    List.map
-      (fun (_, host) -> (host.API.host_uuid, host.API.host_address))
-      (Db.Host.get_all_records ~__context)
+(** Since the liveset info doesn't include the host address, we persist these ourselves *)
+let write_uuid_to_address_mapping ~__context =
+  let uuid_fqdn_of_host (_, host) =
+    let uuid = host.API.host_uuid in
+    let hostname = host.API.host_hostname in
+    match Gencertlib.Lib.fqdn_of_hostname hostname with
+    | None ->
+        raise
+          Api_errors.(
```

Comment:
wouldn't `hostname` be enough in this case? I'm wary about introducing exceptions in the startup process.

Reply:
While testing https://github.com/xapi-project/xen-api/pull/4247/files, I found that stunnel will refuse to connect unless the `checkHost` parameter matches the FQDN (hostname doesn't work), so not throwing here _could_ lead to subtle bugs

I'm not sure how you worked out that this is called in the startup process - I see that it gets used by `preconfigure_ha`, which is presumably only called when enabling ha?

Reply:
i don't know how I reached that conclusion, my bad!